physionet-django/training/models.py

```diff
@@ -0,0 +1,144 @@
+from django.db import models
+
+from project.modelcomponents.fields import SafeHTMLField
+
+
+class Course(models.Model):
+    training_type = models.ForeignKey('user.TrainingType',
+                                      on_delete=models.CASCADE, related_name='courses')
+    version = models.FloatField(default=1.0)
+
+    class Meta:
+        default_permissions = ('change',)
+        unique_together = ('training_type', 'version')
+        permissions = [
+            ('can_view_course_guidelines', 'Can view course guidelines'),
+        ]
+
+    def __str__(self):
+        return f'{self.training_type} v{self.version}'
+
+
+class Module(models.Model):
+    name = models.CharField(max_length=100)
+    course = models.ForeignKey('training.Course', on_delete=models.CASCADE, related_name='modules')
+    order = models.PositiveIntegerField()
+    description = SafeHTMLField()
+
+    class Meta:
+        unique_together = ('course', 'order')
+
+    def __str__(self):
+        return self.name
+
+
+class Quiz(models.Model):
+    question = SafeHTMLField()
+    module = models.ForeignKey('training.Module',
+                               on_delete=models.CASCADE, related_name='quizzes')
+    order = models.PositiveIntegerField()
+
+
+class ContentBlock(models.Model):
+    module = models.ForeignKey('training.Module',
+                               on_delete=models.CASCADE, related_name='contents')
+    body = SafeHTMLField()
+    order = models.PositiveIntegerField()
+
+
+class QuizChoice(models.Model):
+    quiz = models.ForeignKey('training.Quiz',
+                             on_delete=models.CASCADE, related_name='choices')
+    body = models.TextField()
+    is_correct = models.BooleanField('Correct Choice?', default=False)
+
+
+class CourseProgress(models.Model):
+    class Status(models.TextChoices):
+        IN_PROGRESS = 'IP', 'In Progress'
+        COMPLETED = 'C', 'Completed'
+
+    user = models.ForeignKey('user.User', on_delete=models.CASCADE)
+    course = models.ForeignKey('training.Course', on_delete=models.CASCADE)
+    status = models.CharField(max_length=2, choices=Status.choices, default=Status.IN_PROGRESS)
+    started_at = models.DateTimeField(auto_now_add=True)
+    completed_at = models.DateTimeField(auto_now=True)
```

Comment:
```suggestion
    completed_at = models.DateTimeField(null=True, blank=True)
```

... otherwise this will be updated every time the object is saved